californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -271,35 +258,24 @@ public boolean registerOutboundRequestWithTokenOnly(final Exchange exchange) {
 
 	@Override
 	public void remove(final KeyToken token, final Exchange exchange) {
-		boolean removed;
-		int size;
-		synchronized (exchangesByToken) {
-			removed = exchangesByToken.remove(token, exchange);
-			size = exchangesByToken.size();
-		}
+		boolean removed = exchangesByToken.remove(token, exchange);
 		if (removed) {
-			LOGGER.log(Level.FINE, "removing exchange for token {0}, remaining exchanges by tokens: {1}",
-					new Object[] { token, size });
+			LOGGER.log(Level.FINE, "removing exchange for token {0}", new Object[] { token });
```

Comment:
Yes :-). And it's only logged, if the health stuff is enabled.
 
My main concern was the nested `synchronized` in `clear()`

> 		synchronized (messageIdProvider) {
> 			synchronized (exchangesByToken) {

and going over the code, I only saw, that `synchronized` is used to have "atomic logging of the size". So I removed it.